### ISSUE TITLE
Eventhub trigger cardinality not being set in 2.x.x

### ISSUE THREAD

**Issue by Rapol** [serverless/serverless-azure-functions#478]
# This is a Bug Report

## Description

- What went wrong?

Deploying an event hub trigger doesn't set the cardinality in 2.0.9. This worked at least on 1.0.2-16 which we were using before upgrading to 2.x.x.

- What did you expect should have happened?

sls deploy should generate the correct host.json for event hub trigger and include the cardinality

- What was the config you used?

```
processEventHubLogs:
    handler: src/eventHubHandler
    events:
      - eventHub:
        x-azure-settings:
          name: fileMessages
          path: EVENT_HUB_PATH
          connection: EVENT_HUB_CONNECTION_STRING
          cardinality: many
```
## Additional Data

- **_Serverless Framework Version you're using_**: 1.67.0
- **_Serverless CLI Version you're using_**: 2.3.0
- **_Serverless Azure Plugin Version you're using_**: 2.0.9
- **_Operating System_**: mac os

We found that in the default bindings (bindings.json), the binding of type eventhubTrigger cardinality setting was removed in 2.0.0. This setting needs to be added back

```
{
                    "name": "cardinality",
                    "value": "enum",
                    "required": false,
                    "defaultValue": "many",
                    "enum": [
                        {
                            "value": "one",
                            "display": "One"
                        },
                        {
                            "value": "many",
                            "display": "Many"
                        }
                    ],
                    "label": "$eventHubTrigger_cardinality_label",
                    "help": "$eventHubTrigger_cardinality_help"
                }
```

**Comment by tbarlow12** [comment]
Hey @Rapol, I'll see if I can look into this sometime this week. Could you try running without `x-azure-settings`?

```
processEventHubLogs:
    handler: src/eventHubHandler
    events:
      - eventHub:
         name: fileMessages
         path: EVENT_HUB_PATH
         connection: EVENT_HUB_CONNECTION_STRING
         cardinality: many

**Comment by Rapol** [comment]
We just tried it again with and without x-azure-settings and it is adding the cardinality to the host.json in both cases. Not able to reproduce and we tried the same serverless config without any modifications. Closing this, thanks for the quick response.

**Comment by bdatdo0601** [comment]
Hello,

I got into the same issue with version 2.0.13. I was unable to add cardinality into `function.json`

**Comment by Rapol** [comment]
We just migrated to 2.1.0 to be able to deploy node12 and v3 function runtime and stumbled again with this issue. We have our own fork and by adding the following to bindings.json (https://github.com/serverless/serverless-azure-functions/blob/master/src/shared/bindings.json#L60) the plugin set the cardinality correctly in the function.json

```
{
            "type": "eventHubTrigger",
            "displayName": "$eventHubTrigger_displayName",
            "direction": "trigger",
            "enabledInTryMode": false,
            "documentation": "$content=Documentation\\eventHubTrigger.md",
            "settings": [
               // Add cardinality
                {
                    "name": "cardinality",
                    "value": "enum",
                    "required": false,
                    "defaultValue": "many",
                    "enum": [
                        {
                            "value": "one",
                            "display": "One"
                        },
                        {
                            "value": "many",
                            "display": "Many"
                        }
                    ],
                    "label": "$eventHubTrigger_cardinality_label",
                    "help": "$eventHubTrigger_cardinality_help"
                },
```

**Comment by jovanamarko** [comment]
I am having the same issue with 2.1.0 version, seems like the cardinality is set to 'one' even thought the plugin doesn't set it and default should be 'many' as I've read in Azure Event Hub Trigger Function documentation.
This is the definition 
`process_event:
    handler: src/handlers/process_event.processEvent
    events:
      - eventHub: true
        name: eventHubMessages
        eventHubName: EVENT_HUB_NAME_DEV
        consumerGroup: $Default
        connection: EVENT_HUBS_CONNECTION_DEV
        cardinality: many`
even when I'm sending batch of 5 events, the function is executing one time for each, basically instead to get them in an array I get them one by one.

**Comment by ShanonJackson** [comment]
Whats the fix here guys? cardinality seems to be completely ignored in the servleress.yml; And although Azure suggests it should default to "many" its only receiving a single event (not an array).


TL;DR
We set cardinality: many
in Azure there's no config for cardinality after deployment it just disappears

**Comment by Rapol** [comment]
We fixed it by adding the following item to the eventHubTrigger binding settings array

```
                {
                    "name": "cardinality",
                    "value": "enum",
                    "required": false,
                    "defaultValue": "many",
                    "enum": [
                        {
                            "value": "one",
                            "display": "One"
                        },
                        {
                            "value": "many",
                            "display": "Many"
                        }
                    ],
                    "label": "$eventHubTrigger_cardinality_label",
                    "help": "$eventHubTrigger_cardinality_help"
                },
```

Add the previous object to this array https://github.com/serverless/serverless-azure-functions/blob/master/src/shared/bindings.json#L65

**Comment by jovanamarko** [comment]
We have solved this with adding the object in bindings.json file as described by @Rapol, but instead of forking I created a patch using [patch-package](https://www.npmjs.com/package/patch-package). However, this is still an issue that should be considered resolving in next version of the module, even more so because it is a simple fix in bindings.json.

**Comment by Rapol** [comment]
Unfortunately, azure seems to have abandoned this project.  The community might need to fork it to fix bugs and add new features.

Function app has been adding features to hook them up with apim, cdn, authorization, etc somewhat easily so my guess is they want to move to ARM/terraform and not maintain sls framework.
